site/docs/releases.md

```diff
@@ -67,12 +67,29 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
 </dependencies>
 ```
 
+### 1.9.1 release
+
+Apache Iceberg 1.9.1 was released on May 27, 2025.
+
+The 1.9.1 release contains bug fixes and new features. For full release notes visit [Github](https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.9.1)
+
+* API
+    - API, Build: Fix Iceberg Build Version [\#12949](https://github.com/apache/iceberg/pull/12949)
+* Core
+    - Core: Revert [\#12670](https://github.com/apache/iceberg/pull/12670) to Temporarily Restore Sending Single Snapshot Changes Rather than in Bulk [\#13100](https://github.com/apache/iceberg/pull/13100)
+
```

Comment:
I think parquet version bump is to fix CVE and better to include in the release notes? 

Maybe just include all four PR information from diff: 
https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.9.1